.circleci/config.yml

```diff
@@ -0,0 +1,541 @@
+version: 2.1
+
+defaults: &defaults
+  docker:
+    - image: node:6
+  working_directory: ~/snyk-python-plugin
+
+commands:
+  checkout_and_merge:
+    steps:
+      - checkout
+      - run:
+          name: Checkout master
+          command: git checkout origin/master
+      - run:
+          name: Merge test branch
+          command: |
+            git config user.name "CircleCI"
+            git config user.email "noop"
+            git merge --no-edit "$CIRCLE_BRANCH"
+      - attach_workspace:
+          at: ~/snyk-python-plugin
+  test:
+    parameters:
+      pip_ver:
+        type: string
+      python_ver:
+        type: string
+    steps:
+      - checkout_and_merge
+      - attach_workspace:
+          at: ~/snyk-python-plugin
+      - run:
+          name: Run tests
+          command: |
+            apt -qq update
+            apt -qq install python3-pip python-pip -y &> /dev/null
+            curl https://pyenv.run | $SHELL
+            export PATH="/root/.pyenv/bin:$PATH"
+            eval "$(pyenv init -)"
+            eval "$(pyenv virtualenv-init -)"
+            export PYTHON_VER_FULL=`pyenv install --list | grep -v 'Available versions' | awk '{$1=$1};1' | grep "^$PYTHON_VER\.[0-9]\+$" | tail -1`
+            echo $PYTHON_VER_FULL
+            # Install the specific release of Python if it isn't already installed
+            pyenv install -s $PYTHON_VER_FULL
+            pyenv sh-shell $PYTHON_VER_FULL
+            python --version
+            export PATH=$HOME/.local/bin:$PATH
+            python -m pip install --user --quiet pip==$PIP_VER
+            # Call `pyenv rehash` to avoid issues after installing a new version of pip.
+            # see: https://github.com/pyenv/pyenv/issues/1141#issuecomment-383092896
+            pyenv sh-rehash
+            python -m pip --version
+            python -m pip install --user --quiet -r dev-requirements.txt --disable-pip-version-check
```

Comment:
is there a docker image we can use / make to avoid all this?

Reply:
You'd need one per node / python combo, so 8 images? We absolutely could do this / should do this, but its an optimisation for later.

Reply:
This is what we do in the plugin https://github.com/snyk/snyk-python-plugin/blob/master/.travis.yml

Reply:
@lili2311 I dont understand what you're saying :smile:

Reply:
lol, this is the plugin :D

Reply:
oh dear I'm tired 😴